### PR TITLE
WL 21472: restore snails ability to eat sprouts

### PR DESCRIPTION
**Pull request by amvmoody**
Info
- Check with HMD.

# QA

## Setup
- Load in http://hifi-content.s3.amazonaws.com/DomainContent/production/domainsBaked/cupcake/brusselSproutSingle/baked/brusselSproutSingle.baked.fbx 
    - set a colision shape (basic will be fine)
    - replace userdata with
```
{
  "grabbableKey": {
    "cloneLifetime": 300,
    "cloneLimit": 0,
    "cloneable": true,
    "cloneDynamic": true,
    "ignoreIK": false
  },
  "eatable": true
}
```
- Load in http://hifi-content.s3.amazonaws.com/DomainContent/production/domainsBaked/cupcake/snail_Dance2/baked/snail_Dance2.baked.fbx 
    - set a colision shape (basic will be fine)
    -  add this script to the snail https://raw.githubusercontent.com/amvmoody/hifi/dc53017dd4fb68dc63fe7b6c4854057facd63cb6/unpublishedScripts/DomainContent/Cupcake/eatable.js
    - replace userdata with:
```
{
  "grabbableKey": {
    "cloneLifetime": 300,
    "cloneLimit": 0,
    "cloneable": true,
    "cloneDynamic": true
  }
}
```

## Steps - Single Person

| | Steps | Expected results |
| --- | --- | --- |
| 1. | Grab a brussel sprout in HMD | It should clone into your hands |
| 2. | Make the sprout collide with the snail | the snail should make a sound and the spout should be deleted. |

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1919/HighFidelity-Beta-PR11041-dc53017dd4fb68dc63fe7b6c4854057facd63cb6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1919/HighFidelity-Beta-PR11041-dc53017dd4fb68dc63fe7b6c4854057facd63cb6.exe

**Comment by Nex-Pro** [comment]
This I have to test. Gotta love snails.

**Comment by Nex-Pro** [comment]
@amvmoody  Can we have the correct script link please?

**Comment by Nex-Pro** [comment]
I have a hungry and disappointed snail here !

**Comment by amvmoody** [comment]
but the snail wants gravity... (i updated the script link)

**Comment by Nex-Pro** [comment]
@amvmoody Help these snails are starving. They won't eat their greens!

Followed steps.  Used correct script link. There's no interaction between sprout and snail. As if collisions are not being detected.

**Comment by Nex-Pro** [comment]
Both snails and sprouts are cloneable

**Comment by Nex-Pro** [comment]
Works when I run the script locally

**Comment by Nex-Pro** [comment]
### PASS

Ok works now with that new new link.

File: unpublishedScripts/DomainContent/Cupcake/eatable.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    var _entityID;
+
+    this.preload = function (entityID) {
+        _entityID = entityID;
+    };
+
+    this.collisionWithEntity = function(entityUuid, collisionEntityID, collisionInfo) {
+        var entity = Entities.getEntityProperties(collisionEntityID, ['userData', 'name']);
+        var isClone = (entity.name.substring(1).split('-')[0] === 'clone');
+        var isEatable = (JSON.parse(entity.userData).eatable);
```
Comment (zfox23) [inline unpublishedScripts/DomainContent/Cupcake/eatable.js]: Shouldn't this (and other instances of "eatalbe") be "edible" ?

**Pull request merged**